src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/transform/action/get/TransportGetTransformAction.kt

```diff
@@ -0,0 +1,67 @@
+package com.amazon.opendistroforelasticsearch.indexmanagement.transform.action.get
+
+import com.amazon.opendistroforelasticsearch.indexmanagement.IndexManagementPlugin.Companion.INDEX_MANAGEMENT_INDEX
+import com.amazon.opendistroforelasticsearch.indexmanagement.elasticapi.parseWithType
+import com.amazon.opendistroforelasticsearch.indexmanagement.transform.model.Transform
+import org.elasticsearch.ElasticsearchStatusException
+import org.elasticsearch.action.ActionListener
+import org.elasticsearch.action.get.GetRequest
+import org.elasticsearch.action.get.GetResponse
+import org.elasticsearch.action.support.ActionFilters
+import org.elasticsearch.action.support.HandledTransportAction
+import org.elasticsearch.client.Client
+import org.elasticsearch.common.inject.Inject
+import org.elasticsearch.common.xcontent.LoggingDeprecationHandler
+import org.elasticsearch.common.xcontent.NamedXContentRegistry
+import org.elasticsearch.common.xcontent.XContentHelper
+import org.elasticsearch.common.xcontent.XContentType
+import org.elasticsearch.rest.RestStatus
+import org.elasticsearch.tasks.Task
+import org.elasticsearch.transport.TransportService
+
+class TransportGetTransformAction @Inject constructor(
+    transportService: TransportService,
+    val client: Client,
+    actionFilters: ActionFilters,
+    val xContentRegistry: NamedXContentRegistry
+) : HandledTransportAction<GetTransformRequest, GetTransformResponse> (
+    GetTransformAction.NAME, transportService, actionFilters, ::GetTransformRequest
+) {
+
+    override fun doExecute(task: Task, request: GetTransformRequest, listener: ActionListener<GetTransformResponse>) {
+        val getRequest = GetRequest(INDEX_MANAGEMENT_INDEX, request.id)
+            .fetchSourceContext(request.srcContext).preference(request.preference)
+        client.get(getRequest, object : ActionListener<GetResponse> {
+            override fun onResponse(response: GetResponse) {
+                if (!response.isExists) {
+                    return listener.onFailure(ElasticsearchStatusException("Transform not found", RestStatus.NOT_FOUND))
+                }
+
+                var transform: Transform? = null
+                if (!response.isSourceEmpty) {
```

Comment:
Looks like this happens if the index we are searching from has the `_source` disabled which shouldn't happen here since the system index where we store these docs have the source enabled. I think it would be better to throw an Internal exception instead if source is empty instead of returning empty transform object.

https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-source-field.html

Reply:
Looks like we are doing a ``HEAD`` REST call which doesn't return the actual transform but just the id and primary/seq no. For now ignore my comment. 

We should not throw the exception but rather just return empty in those cases

Reply:
Updated to include this logic and also properly handle HEAD calls.